conda/models/match_spec.py

```diff
@@ -183,7 +183,7 @@ def from_dist_str(cls, dist_str):
         if '::' in dist_str:
             channel_subdir_str, dist_str = dist_str.split("::", 1)
             if '/' in channel_subdir_str:
-                channel_str, subdir = channel_subdir_str.split('/', 2)
+                channel_str, subdir = channel_subdir_str.rsplit('/', 1)
```

Comment:
`subdir` needs to be checked against `context.known_subdirs`. If `subdir` isn't in `known_subdirs`, then `subdir` should be set to `None` and everything goes into `channel_str`.

Reply:
Also should include both situations in the unit tests

Reply:
I'll make those changes, thank you!